Detectors/Base/include/DetectorsBase/MaterialManager.h

```diff
@@ -204,17 +204,16 @@ class MaterialManager
   // print out all registered media
   void printMedia() const;
 
-  /// set the density scaling factor
-  void setDensityScalingFactor(float f) { mDensityFactor = f; }
-  /// set the density scaling factor
-  float getDensityScalingFactor() const { return mDensityFactor; }
-
   /// print all tracking media inside a logical volume (specified by name)
   /// and all of its daughters
   static void printContainingMedia(std::string const& volumename);
 
  private:
-  MaterialManager() = default;
+  std::unordered_map<std::string, float> mDensityMap;
+
+  void createDensityMap();
+
+  MaterialManager() { createDensityMap(); }
```

Comment:
not sure this is a good idea. Initialization might happen during construction of the singleton MaterialManager, at which moment the ConfigParams might not be fully initialized.

It may be better to init this *late*. For instance we could have a function 
```
float getDensity(std::string const& modname) {
  if (!initialized) {
     initDensityMap();
  }
  if (mDensityMap.find(modname) != mDensityMap.end()) {
    return mDensityMap[modname]
  }
  return mGlobalDensityFactor;
}
```

Reply:
The suggested function has been implemented in the code.